src/include/miopen/hipoc_kernel.hpp

```diff
@@ -65,9 +65,16 @@ struct KernelArgsPair
 template <class T, class U>
 struct KernelArgsPair
 {
-    KernelArgsPair(T x, U y) : first(x), second(y) {}
-    T first;
-    U second;
+    static const int alignment = alignof(U);
+    static const int padding   = (alignment - (sizeof(T) % alignment)) % alignment;
+    static_assert(padding >= 0, "padding cannot be negative");
```

Comment:
I will revert the `static_assert` however, there's another subtle bug that I fixed here and that has to do with line 68 that needs `alignment = alignof(U)` instead of `alignment = sizeof(U)`. This was causing kernel arguments to be passed with wrong amount of padding and thus garbled during execution.

Reply:
@amberhassaan 👍 [resolved]